src/systems/ackermann_steering/AckermannSteering.cc

```diff
@@ -301,77 +329,98 @@ void AckermannSteering::Configure(const Entity &_entity,
     this->dataPtr->limiterAng->SetMaxJerk(maxJerk);
   }
 
-
-  double odomFreq = _sdf->Get<double>("odom_publish_frequency", 50).first;
-  if (odomFreq > 0)
+  if (!this->dataPtr->steeringOnly)
   {
-    std::chrono::duration<double> odomPer{1 / odomFreq};
-    this->dataPtr->odomPubPeriod =
-      std::chrono::duration_cast<std::chrono::steady_clock::duration>(odomPer);
+    double odomFreq = _sdf->Get<double>("odom_publish_frequency", 50).first;
+    if (odomFreq > 0)
+    {
+      std::chrono::duration<double> odomPer{1 / odomFreq};
+      this->dataPtr->odomPubPeriod =
+        std::chrono::duration_cast<std::chrono::steady_clock::duration>(
+          odomPer);
+    }
   }
-
   // Subscribe to commands
   std::vector<std::string> topics;
   if (_sdf->HasElement("topic"))
   {
     topics.push_back(_sdf->Get<std::string>("topic"));
   }
-  topics.push_back("/model/" + this->dataPtr->model.Name(_ecm) + "/cmd_vel");
-  auto topic = validTopic(topics);
-  if (topic.empty())
+  if (this->dataPtr->steeringOnly)
   {
-    gzerr << "AckermannSteering plugin received invalid model name "
-           << "Failed to initialize." << std::endl;
-    return;
+    topics.push_back("/model/" + this->dataPtr->model.Name(_ecm) +
+      "/steer_angle");
   }
-
-  this->dataPtr->node.Subscribe(topic, &AckermannSteeringPrivate::OnCmdVel,
-      this->dataPtr.get());
-
-  std::vector<std::string> odomTopics;
-  if (_sdf->HasElement("odom_topic"))
+  else
   {
-    odomTopics.push_back(_sdf->Get<std::string>("odom_topic"));
+    topics.push_back("/model/" + this->dataPtr->model.Name(_ecm) + "/cmd_vel");
   }
-  odomTopics.push_back("/model/" + this->dataPtr->model.Name(_ecm) +
-      "/odometry");
-  auto odomTopic = validTopic(odomTopics);
+
+  auto topic = validTopic(topics);
   if (topic.empty())
   {
     gzerr << "AckermannSteering plugin received invalid model name "
            << "Failed to initialize." << std::endl;
     return;
   }
-
-  this->dataPtr->odomPub = this->dataPtr->node.Advertise<msgs::Odometry>(
-      odomTopic);
-
-  std::vector<std::string> tfTopics;
-  if (_sdf->HasElement("tf_topic"))
+  if (this->dataPtr->steeringOnly)
   {
-    tfTopics.push_back(_sdf->Get<std::string>("tf_topic"));
+    this->dataPtr->node.Subscribe(topic, &AckermannSteeringPrivate::OnCmdAng,
+      this->dataPtr.get());
+    gzmsg << "AckermannSteering subscribing to float messages on [" <<
+      topic << "]" << std::endl;
   }
-  tfTopics.push_back("/model/" + this->dataPtr->model.Name(_ecm) +
-    "/tf");
-  auto tfTopic = validTopic(tfTopics);
-  if (tfTopic.empty())
+  else
   {
-    gzerr << "AckermannSteering plugin invalid tf topic name "
-           << "Failed to initialize." << std::endl;
-    return;
+    this->dataPtr->node.Subscribe(topic, &AckermannSteeringPrivate::OnCmdVel,
+      this->dataPtr.get());
+    gzmsg << "AckermannSteering subscribing to twist messages on [" <<
+      topic << "]" << std::endl;
```

Comment:
```suggestion
    gzmsg << "AckermannSteering subscribing to twist messages on [" 
          << topic << "]" << std::endl;
```

Reply:
Should be good now.

Reply:
Actually found another style like that in it and fixed it to match.